hazelcast-it/jdk17-tests/src/test/java/com/hazelcast/serialization/compact/record/AllTypesRecord.java

```diff
@@ -72,7 +75,10 @@ public record AllTypesRecord(
         LocalDateTime[] timestampArray,
         OffsetDateTime[] timestampWithTimezoneArray,
         AnEnum[] anEnumArray,
-        NestedRecord[] nestedRecordArray
+        NestedRecord[] nestedRecordArray,
+        ArrayList<String> arrayList,
```

Comment:
Can we add test for `List, Map, Set` field inside ?

Reply:
Done in https://github.com/hazelcast/hazelcast/pull/21980/commits/b036e04d4fe309e1330c4c1f9cc89ee2914dc29e